backends-velox/src/main/scala/org/apache/gluten/execution/ColumnarPartialProjectExec.scala

```diff
@@ -172,6 +172,20 @@ case class ColumnarPartialProjectExec(original: ProjectExec, child: SparkPlan)(
     }
   }
 
+  private def toVeloxBatch(input: ColumnarBatch): ColumnarBatch = {
+    if (VeloxColumnarBatches.isVeloxBatch(input)) {
+      return input
+    }
+    if (ColumnarBatches.isHeavyBatch(input)) {
+      if (input.numCols() == 0) {
+        return input
+      }
+      return VeloxColumnarBatches.toVeloxBatch(
+        ColumnarBatches.offload(ArrowBufferAllocators.contextInstance(), input))
+    }
```

Comment:
```
If (batch is not in velox format) {
  convert batch to velox
}
```

Such logics are considered implicit transitions and were removed in effort of https://github.com/apache/incubator-gluten/issues/7313. 

Operators now have certain input formats so we can make clear assumption that which batch is in which format. So let's remove these implicit transitions. Thanks.

Reply:
The ArrowCsvReader may produce Arrow format heavy ColumnarBatch, does it work?

Reply:
For the plan ProjectExec, child plan is csv scan.

Reply:
> The ArrowCsvReader may produce Arrow format heavy ColumnarBatch, does it work?

It should work. Planner should take the work to insert Arrow-to-Velox transitions now, you can run some tests to see the final plan Gluten gives.